i686 libstdc++-6.dll built after latest mingw crt/winpthreads/etc update breaks things

After installing a freshly built gcc, things are silently failing. cmake was the first thing I saw, but then when I tried to find out what was happening I saw gdb was also silently failing.
the most likely thing that changed between when it worked and now would be the mingw crt/etc packages were updated

dependency walker is showing libstdc++ has undefined imports `__emutls_v._ZSt11_once_call` and `__emutls_v.Zst15_once_callable`. Those symbols are present in the existing/working libstdc++-6.dll.

libstdc++-6.dll from CI rebuild of gcc has the issue. Local build of gcc after downgrading mingw packages(`pacman -U /var/cache/pacman/pkg/mingw-w64-i686-*-8.0.0.5*`) ongoing, but there is already a stage3 libstdc++-6.dll, which defines those symbols.

Same symbol:
```
configure:28821: /home/adminuser/MINGW-packages/mingw-w64-gcc/src/build-i686-w64-mingw32/./gcc/xgcc -B/home/adminuser/MINGW-packages/mingw-w64-gcc/src/build-i686-w64-mingw32/./gcc/ -L/mingw32/i686-w64-mingw32/lib -L/mingw32/lib -isystem /mingw32/i686-w64-mingw32/include -isystem /mingw32/include -B/mingw32/i686-w64-mingw32/bin/ -B/mingw32/i686-w64-mingw32/lib/ -isystem /mingw32/i686-w64-mingw32/include -isystem /mingw32/i686-w64-mingw32/sys-include   -fno-checking -o conftest.exe -g -march=i686 -mtune=generic -O2 -pipe  -static -pipe -Wl,--dynamicbase,--nxcompat,--no-seh -Wl,--large-address-aware conftest.c  >&5
C:\msys64\mingw32\i686-w64-mingw32\bin\ld.exe: C:/msys64/mingw32/i686-w64-mingw32/lib\libmingwex.a(lib32_libmingwex_a-mingw_pformat.o): in function `_pformat_int':
C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/stdio/mingw_pformat.c:797: undefined reference to `__umoddi3'
collect2.exe: error: ld returned 1 exit status
configure:28821: $? = 1
```

--- a/mingw-w64-winpthreads-git/PKGBUILD
+++ b/mingw-w64-winpthreads-git/PKGBUILD
@@ -6,7 +6,7 @@ _realname=winpthreads
 pkgbase=mingw-w64-${_realname}-git
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}-git" "${MINGW_PACKAGE_PREFIX}-libwinpthread-git")
 pkgver=8.0.0.6001.98dad1fe
-pkgrel=2
+pkgrel=3
 _commit='98dad1fefed2a07ade0b37ca6c610ad66739631a'
 pkgdesc="MinGW-w64 winpthreads library"
 url="https://mingw-w64.sourceforge.io/"
@@ -49,6 +49,8 @@ build() {
   mkdir -p "${srcdir}"/build-${MINGW_CHOST} && cd ${srcdir}/build-${MINGW_CHOST}
 
   declare -a extra_config
+  # https://github.com/msys2/MINGW-packages/issues/7043
+  extra_config+=("CPPFLAGS=${CPPFLAGS//-D__USE_MINGW_ANSI_STDIO=1/}")
   if check_option "debug" "y"; then
     extra_config+=("CFLAGS=-O0 -g -DWINPTHREAD_DBG")
     extra_config+=("CXXFLAGS=-O0 -g -DWINPTHREAD_DBG")

